File: 3. TESTE DE BANCO DE DADOS/test_database_python/scripts/data_config.py
import os
import json
from dotenv import load_dotenv


class DataConfig:
    def __init__(self):
        load_dotenv()

        base_dir = os.path.dirname(__file__)
        config_dir = os.path.join(base_dir, "..", "config")

        self.config_path = os.path.join(config_dir, "config.json")
        self.import_csv_data = os.path.join(config_dir, "import_csv_data.json")

        with open(self.config_path, "r", encoding="utf-8") as f:
            self.general_config = json.load(f)

        with open(self.import_csv_data, "r", encoding="utf-8") as f:
            self.script_config = json.load(f)

    def get_sources(self):
        return self.general_config

    def get_paths(self):
        return self.script_config.get("paths", {})

    def get_mysql_config(self):
        return {
            "host": os.getenv("MYSQL_HOST"),
            "port": int(os.getenv("MYSQL_PORT")),
            "user": os.getenv("MYSQL_USER"),
            "password": os.getenv("MYSQL_PASSWORD"),
            "database": os.getenv("MYSQL_DATABASE"),
            "allow_local_infile": True,
        }

File: 3. TESTE DE BANCO DE DADOS/test_database_java/src/main/java/com/test/database/config/DataConfig.java
package com.test.database.config;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.test.database.LogsMain;
import io.github.cdimascio.dotenv.Dotenv;

import java.io.FileReader;
import java.lang.reflect.Type;
import java.nio.file.Paths;
import java.util.List;
import java.util.Map;

public class DataConfig {

    private final List<Map<String, Object>> sources;
    private final String url;
    private final String user;
    private final String password;

    public DataConfig() {
        Dotenv dotenv = Dotenv.load();

        String host = dotenv.get("MYSQL_HOST", "localhost");
        String port = dotenv.get("MYSQL_PORT", "3306");
        String database = dotenv.get("MYSQL_DATABASE", "ans_database");
        this.user = dotenv.get("MYSQL_USER", "root");
        this.password = dotenv.get("MYSQL_PASSWORD", "");

        this.url = "jdbc:mysql://" + host + ":" + port + "/" + database +
                "?useSSL=false&serverTimezone=UTC&allowLoadLocalInfile=true";

        try {
            String path = Paths.get("config", "config.json").toString();
            FileReader reader = new FileReader(path);
            Type listType = new TypeToken<List<Map<String, Object>>>() {}.getType();
            this.sources = new Gson().fromJson(reader, listType);
        } catch (Exception e) {
            LogsMain.logError("Erro ao carregar config.json: " + e.getMessage());
            throw new RuntimeException("Erro ao carregar config.json", e);
        }
    }

    public String getUrl() {
        return url;
    }

    public String getUser() {
        return user;
    }

    public String getPassword() {
        return password;
    }

    public List<Map<String, Object>> getSources() {
        return sources;
    }
}
